src/app/add-clients/pageComponents.tsx

```diff
@@ -0,0 +1,740 @@
+                quantity: quantity,
+                age: null,
+            };
+            const { error: error } = await supabase.from("families").insert(familyAdultRecord);
+            if (error) {
+                console.error(error);
+            }
+        }
+
+        for (const child of ageGenderChildren) {
+            const familyChildRecord: FamilyDatabaseRecord = {
+                family_id: familyID![0].family_id,
+                person_type: child.gender as PersonType,
+                quantity: 1,
+                age: child.age,
+            };
+            const { error: error } = await supabase.from("families").insert(familyChildRecord);
+            if (error) {
+                console.error(error);
+            }
+        }
+    };
+
+    const submitForm = (): void => {
+        let errorExists = false;
+        let amendedErrorMessages = { ...errorMessages };
+        for (const [errorKey, errorMessage] of Object.entries(errorMessages)) {
+            if (errorMessage !== "") {
+                errorExists = true;
+            }
+            if (errorMessage === "N/A") {
+                amendedErrorMessages = {
+                    ...amendedErrorMessages,
+                    [errorKey]: "This is a required field.",
+                };
+            }
+        }
+        if (errorExists) {
+            setErrorMessages({ ...amendedErrorMessages });
+            alert(
+                "Please ensure all fields have been entered correctly. Required fields are labelled with an asterisk."
+            );
+        } else {
+            void createRecord();
+        }
+    };
+
+    const errorExists = (errorMessage: string): boolean => {
+        return errorMessage !== "" && errorMessage !== "N/A";
+    };
+
+    const errorText = (errorMessage: string): string => {
+        return errorMessage == "N/A" ? "" : errorMessage;
+    };
+
+    return (
+        <CenterComponent>
+            <StyledForm>
+                <Heading>Client Form</Heading>
+                <Text>
+                    Please provide or update the client&apos;s personal details, household
+                    composition, dietary restrictions and other needs.
+                </Text>
+                <CenterComponent>
+                    <StyledCard>
+                        <Subheading>
+                            Client Full Name <Asterisk />
+                        </Subheading>
+                        <Text>First and last name</Text>
+                        <FreeFormTextInput
+                            error={errorExists(errorMessages.nameErrorMessage)}
+                            helperText={errorText(errorMessages.nameErrorMessage)}
+                            label="Name"
+                            onChange={getRequiredFieldWithoutChecks(
+                                "nameErrorMessage",
+                                setFullName
+                            )}
+                        />
```

Comment:
For example, this code can be taken out into a separate component called `<ClientFullNameSection />` or something like that. You would also have to move the corresponding functions specific to the section to the new component file as well